clang/lib/CodeGen/Targets/X86.cpp

```diff
@@ -3014,6 +3014,11 @@ Address X86_64ABIInfo::EmitVAArg(CodeGenFunction &CGF, Address VAListAddr,
   ABIArgInfo AI = classifyArgumentType(Ty, 0, neededInt, neededSSE,
                                        /*isNamedArg*/false);
 
+  // Empty records are ignored for parameter passing purposes.
+  if (AI.isIgnore()) {
+    return CGF.CreateMemTemp(Ty);
+  }
```

Comment:
Does it behave differently between C and C++? Maybe adding a C test for it?

Besides, do not use parentheses for single line code.